### DISCUSSION TITLE
Changed the name

### DISCUSSION THREAD

**Discussion by Gaspari11** [small-cactus/M.I.L.E.S#17] [Q&A] [answered]
Hi I’m just wondering if there is a way to change the name from miles to something different I tried editing the main.py and the config.py but no luck

**Comment by small-cactus** [comment, marked as answer]
The wake word detection for Miles is trained on 50,000 samples of people saying “Miles”. It’s possible to train another wake word model for another name but it’s gonna take anywhere from 30 minutes, to 6 hours to train it.

the model for Miles took 6 hours to train on an A100 GPU via colab for a reference.

Here’s the link to train the AI model [link](https://colab.research.google.com/drive/1q1oe2zOyZp7UsB3jJiQ1IFn8z5YfjwEb?usp=sharing)

You should just be able to follow the instructions, you type a name where it says khum_puter, it can be anything, click the play button, once it’s done make sure it sounds right, then go through the other cards and click play until it’s done. Once it’s done you’ll need to place the file inside the folder named just “miles”.

from here you have 2 choices, either rename the newly trained file to miles-50k, or edit main.py, searching for miles-50k and change it to the name of the file you trained.


If you have any other questions or need help training it, I can help you.

**Reply by Gaspari11** [reply]
I followed the instructions, the Ai trained and gave me the 2 files i changed all occurrences of miles in the main.py file i added the 2 new files into the miles folder but it only seems to respond to miles is there anything else i need to changed or do ? thx for you help

**Reply by small-cactus** [reply]
You shouldn't change all occurrences of miles, that breaks the app because several parts of the frontend rely on the name Miles at the beginning of responses to display UI elements, you should change all instances of miles-50k to the name of the file it gave you.

miles-50k.onnx

miles-50k.tflite

These are the files that house the ai models for wake word detection. They are also the same name inside the main.py file.

**Reply by Gaspari11** [reply]
it didnt break the app which is good, i did also change the miles-50k to my file name there were 6 of them and i added the 2 new files to the miles folder showing the new name so what do you think i missed

**Reply by small-cactus** [reply]
Ah I see what you mean, I forgot I added makeshift linux support, my development copy isnt updated fully. The way you made it should work, may I ask what settings you used for it? Did you use the default ones? It may just be inaccurate or too high of a threshold. Try changing the value for `THRESHOLD` to a number that more fits your model. You might have to try 0.5 or 0.1, or any other value that works for you.

**Comment by Gaspari11** [comment]
Thank you very much for your well detailed answer
im trying it now im not a coder so it takes me longer but im learning i may
have some more questions for you

Thanks for your fast reply
